src/main.rs

```diff
@@ -420,11 +422,26 @@ fn run() -> Result<ExitStatus> {
         }
     }
 
-    eprintln!("Warning: Falling back to `cargo` on the host.");
-
     // if we fallback to the host cargo, use the same invocation that was made to cross
     let argv: Vec<String> = env::args().skip(1).collect();
-    cargo::run(&argv, verbose)
+    eprintln!("Warning: Falling back to `cargo` on the host.");
+    match args.subcommand {
+        Some(Subcommand::List) => {
+            // this won't print in order if we have both stdout and stderr.
+            let out = cargo::run_and_get_output(&argv, verbose)?;
+            let stdout = out.stdout()?;
+            if out.status.success() && cli::is_subcommand_list(&stdout) {
+                cli::fmt_subcommands(&stdout);
+            } else {
+                // Not a list subcommand, which can happen with weird edge-cases.
+                print!("{}", stdout);
+                io::stdout().flush().unwrap();
+            }
+            Ok(out.status)
+        }
+        None | Some(Subcommand::Other) => cargo::run(&argv, verbose),
```

Comment:
Should be the universal match arm, since we can fall back to the host cargo if the image doesn't exist or if the target isn't supported (the two failing examples above).

```rust
_ => cargo::run(&argv, verbose),
```